### PR TITLE
f-header@v4.4.0 - Updated feature tests with selector.js file

### PR DESCRIPTION
**Pull request by LTurns**
v4.4.0
------------------------------
*December 18, 2020*

### Changed
- Updated Webdriver test-utils to include `selector` file.
- Refactored component tests.

Updated beforeHook to take in a new storybook URL which can then be edited to include add-ons - happy to discuss this if people prefer the original way, but it seemed to make testing slightly easier. :)

### CODE REVIEW

**Review by simonsandersje — APPROVED** [review]

File: packages/f-header/test-utils/component-objects/f-header.component.js [hunk 1]
```diff
@@ -1,57 +1,58 @@
...
+const headerComponent = () => $(HEADER_COMPONENT);
+
+// Logo
+const headerLogo = () => $(HEADER_LOGO);
+
+// Navigation
+const navigation = {
+    offers: {
+        link: () => $(NAVIGATION.offers.link), 
+        icon: () => $$(NAVIGATION.offers.link)
```
Comment (siggerzz) [inline packages/f-header/test-utils/component-objects/f-header.component.js:18]: Should this be `NAVIGATION.offers.icon`?

File: packages/f-header/test/specs/component/f-header.component.spec.js [hunk 2]
```diff
@@ -1,85 +1,71 @@
...
-        expect(HeaderComponent.isHelpLinkDisplayed()).toBe(true);
-        expect(HeaderComponent.isLoginLinkDisplayed()).toBe(true);
+        expect(HeaderComponent.isFieldLinkDisplayed(field)).toBe(true);
     });
 
-    it.skip('should change url when help link is clicked', () => {
+    it('should alter visibility of navbar depending on window size', () => {
         // Act
-        HeaderComponent.clickHelpLink();
+        browser.setWindowSize(500, 500);
```
Comment (siggerzz) [inline packages/f-header/test/specs/component/f-header.component.spec.js:34]: Nit: Might be worth refactoring this at some point to use the built-in Storybook viewport feature

File: packages/f-header/test/specs/component/f-header.component.spec.js [hunk 3]
```diff
@@ -1,85 +1,71 @@
...
+        expect(HeaderComponent.isMobileNavigationBarVisible()).toBe(true);
+        expect(HeaderComponent.isMobileOffersIconDisplayed()).toBe(true);
+        expect(HeaderComponent.isFieldLinkDisplayed('help')).toBe(false);
     });
 
-    it.skip('should change url when login link is clicked', () => {
+    it('should change url when help-link is clicked', () => {
         // Act
-        HeaderComponent.clickLoginLink();
+        browser.setWindowSize(1000, 1000);
```
Comment (siggerzz) [inline packages/f-header/test/specs/component/f-header.component.spec.js:44]: ditto

File: packages/f-header/test/specs/component/f-header.component.spec.js [hunk 4]
```diff
@@ -1,85 +1,71 @@
...
-        // resize to mobile
-        browser.setWindowSize(500, 500);
-
-        expect(HeaderComponent.isNavigationOffersLinkDisplayed()).toBe(false);
-        expect(HeaderComponent.isMobileOffersIconDisplayed()).toBe(true);
-
+    forEach(['offers', 'help', 'delivery', 'userAccount'])
+    .it.skip('should show navigation fields when mobile-navigation has been opened', field => {
         // Act
+        browser.setWindowSize(500, 1000);
```
Comment (siggerzz) [inline packages/f-header/test/specs/component/f-header.component.spec.js:62]: ditto

**Review by siggerzz — CHANGES_REQUESTED** [review]
Just a few comments 👍

File: packages/f-header/test/specs/component/f-header.component.spec.js (hunk 2 above)
Comment (LTurns) [reply packages/f-header/test/specs/component/f-header.component.spec.js:34]: ooh yes great shout

File: packages/f-header/test-utils/component-objects/f-header.component.js (hunk 1 above)
Comment (LTurns) [reply packages/f-header/test-utils/component-objects/f-header.component.js:18]: ah so to grab the icon, I'm using the same data-test-id as link, but just iterating over it with $$, which is why I didn't add icon to the NAVIGATION object. Happy to refactor later down the line if you think it'll be clearer though?
I'm on holiday too now so might be better looking into as a separate PR and getting this one merged before Ash makes changes to the folder structure 😄

**Review by ashleynolan — CHANGES_REQUESTED** [review]
Marking this as do not merge. This PR will need refactoring once the folder structure is changed (probably easier to make a copy of the changes you have made and re-apply them to the new locations, as merging will likely give you a bunch of conflicts)

**Comment by LTurns** [comment]
> Marking this as do not merge. This PR will need refactoring once the folder structure is changed (probably easier to make a copy of the changes you have made and re-apply them to the new locations, as merging will likely give you a bunch of conflicts)

perfect, thanks @ashleynolan, will do this today 😄

**Pull request closed without merging**